mono/mini/debug-mini.c

```diff
@@ -616,13 +621,18 @@ mono_debug_add_aot_method (MonoDomain *domain, MonoMethod *method, guint8 *code_
 	if (debug_info_len == 0)
 		return;
 
-	jit = deserialize_debug_info (method, code_start, debug_info, debug_info_len);
-
-	mono_debug_add_method (method, jit, domain);
-
-	mono_debug_add_vg_method (method, jit);
-
-	mono_debug_free_method_jit_info (jit);
+	/* Cache the data here, and decode it when its needed in mini_debug_find_jit_debug_info () */
+	AotDebugInfo *info = mono_domain_alloc0 (domain, sizeof (AotDebugInfo));
+	info->code_start = code_start;
+	info->debug_info = debug_info;
+	info->debug_info_len = debug_info_len;
+
+	domain_info = domain_jit_info (domain);
+	mono_domain_lock (domain);
+	if (!domain_info->aot_debug_info_hash)
+		domain_info->aot_debug_info_hash = g_hash_table_new_full (mono_aligned_addr_hash, NULL, NULL, g_free);
```

Comment:
Should this free memory returned by mono_domain_alloc0?

Reply:
Fixed.